packages/framer-motion/src/render/html/utils/build-transform.ts

```diff
@@ -18,22 +21,43 @@ const numTransforms = transformPropOrder.length
  * providing a transformTemplate function.
  */
 export function buildTransform(
+    latestValues: ResolvedValues,
     transform: HTMLRenderState["transform"],
-    transformIsDefault: boolean,
     transformTemplate?: MotionProps["transformTemplate"]
 ) {
     // The transform string we're going to build into.
     let transformString = ""
+    let transformIsDefault = true
 
     /**
      * Loop over all possible transforms in order, adding the ones that
      * are present to the transform string.
      */
     for (let i = 0; i < numTransforms; i++) {
         const key = transformPropOrder[i] as keyof typeof translateAlias
-        if (transform[key] !== undefined) {
+        const value = latestValues[key]
+
+        if (value === undefined) continue
+
+        let valueIsDefault = true
+        if (typeof value === "number") {
+            valueIsDefault = value === (key.startsWith("scale") ? 1 : 0)
+        } else {
+            valueIsDefault = parseFloat(value) === 0
+        }
+
+        if (!valueIsDefault) {
+            transformIsDefault = false
+        }
+
+        if (!valueIsDefault || transformTemplate) {
             const transformName = translateAlias[key] || key
-            transformString += `${transformName}(${transform[key]}) `
+            const valueAsType = getValueAsType(value, numberValueTypes[key])
```

Comment:
Is this a just a drive by to use a new shared helper?

Reply:
No this was moved in from `build-styles` so we basically only run it if the transform isn't default or if we need to pass it to `transformTemplate`